src/main/java/org/ice4j/ice/Component.java

```diff
@@ -902,6 +922,12 @@ public RemoteCandidate findRemoteCandidate(TransportAddress remoteAddress)
      */
     protected void setSelectedPair(CandidatePair pair)
     {
+        if (keepAliveStrategy == KeepAliveStrategy.SELECTED_ONLY)
```

Comment:
since the keep-alive strategy can be changed at any time, i'm wondering if maybe it makes sense to keep track of all pairs and then just denote one of them as 'selected'?  that way we don't lose track of any of them.  might be a rare thing that the strategy is changed mid-way through, but does look like it's technically possible?